src/content/app/entity-viewer/variant-view/transcript-consequences/transcript-variant-cds/TranscriptVariantCDS.tsx

```diff
@@ -0,0 +1,323 @@
+import type { ExonWithRelativeLocationInCDS } from 'src/shared/helpers/exon-helpers/exonHelpers';
+
+import styles from './TranscriptVariantCDS.module.css';
+
+type ExonFields = Pick<SplicedExon, 'index'> &
+  Pick2<SplicedExon, 'relative_location', 'start' | 'end'>;
+type CDSFields = Pick<
+  FullCDS,
+  'relative_start' | 'relative_end' | 'nucleotide_length'
+>;
+type VariantRelativeLocationFields = Pick<
+  VariantRelativeLocation,
+  'start' | 'end'
+>;
+
+/**
+ * WARNING!
+ * This component is using spliced exons from the core api to draw a CDS diagram.
+ * This will fail to work if we ever get cases of trans-splicing.
+ * It should be api's responsibility to tell the client how exons are arranged along a CDS.
+ */
+
+type Props = {
+  exons: Array<ExonFields & ExonWithRelativeLocationInCDS>;
+  cds: CDSFields;
+  allele: {
+    type: string;
+    relativeLocation: VariantRelativeLocationFields;
+  };
+};
+
+const DIAGRAM_WIDTH = 700;
+const MIN_EXON_BLOCK_WIDTH = 2;
+const EXON_MARGIN_WIDTH = 1;
+const EXON_BLOCK_HEIGHT = 12;
+const MIN_EXON_BLOCK_WITH_ARROW_WIDTH = 18;
+const VARIANT_MARKER_HEIGHT = 18;
+const VATIANT_MARKER_WIDTH = 2;
+const DIAGRAM_HEIGHT = 42;
+const EXON_TO_LABEL_DISTANCE = 12; // distance between the bottom of exon blocks and the top of the label under the variant mark
+
+const EXON_BLOCK_OFFSET_TOP = (VARIANT_MARKER_HEIGHT - EXON_BLOCK_HEIGHT) / 2;
+
+const TranscriptVariantCDS = (props: Props) => {
+  const { exons, allele, cds } = props;
+
+  const scale = scaleLinear()
+    .domain([1, cds.nucleotide_length])
+    .range([0, DIAGRAM_WIDTH])
+    .interpolate(interpolateRound)
+    .clamp(true);
+
+  const exonsWithWidths = getExonWidths({
+    exons,
+    containerWidth: DIAGRAM_WIDTH,
+    scale
+  });
+
+  return (
+    <svg
+      width={DIAGRAM_WIDTH}
+      height={DIAGRAM_HEIGHT}
+      viewBox={`0 0 ${DIAGRAM_WIDTH} ${DIAGRAM_HEIGHT}`}
+      overflow="visible"
+    >
+      <Exons exons={exonsWithWidths} />
+      <VariantMark
+        exons={exonsWithWidths}
+        allele={allele}
+        scale={scale}
+        containerWidth={DIAGRAM_WIDTH}
+      />
+    </svg>
+  );
+};
+
+const Exons = (props: { exons: Array<{ width: number }> }) => {
+  let x = 0;
+
+  const exonBlocks = props.exons.map((exon, index) => {
+    const width = exon.width;
+    const currentX = x;
+    x += width + EXON_MARGIN_WIDTH;
+
+    return <ExonBlock key={index} x={currentX} width={width} />;
+  });
+
+  return <g>{exonBlocks}</g>;
+};
+
+const ExonBlock = (props: { x: number; width: number }) => {
+  const { x, width } = props;
+  const initialChevronWidth = 32; // <-- from the viewBox attribute of the svg image, in which the chevron is pointing down
+  const initialChevronHeight = 20; // <-- from the viewBox attribute of the svg image, in which the chevron is pointing down
+  const targetChevronWidth = 9; // <-- before rotating the chevron 90°; after the rotation this will become its height
```

Comment:
Can't these values be turned into constants like above?

Reply:
They are local to a single function, and to a rather minor concern of chevron dimensions. I'm not sure it's worth  there'll be any benefit